tests/main/snap-icons-fallback/task.yaml

```
@@ -44,41 +78,41 @@ execute: |
     echo "Check that the snap which is NOT supposed to have a local icon does not have no local icon"
     find "${SNAP_DIR}/${SNAP_FALLBACK}"/current/meta/gui/icon.* | wc -l | MATCH 0
 
-
     echo "Check that both snaps have store icons"
     snap debug api "/v2/snaps/$SNAP_LOCAL" | MATCH '"icon"'
     snap debug api "/v2/snaps/$SNAP_FALLBACK" | MATCH '"icon"'
 
-
-    echo "Check that installed snaps now have icons downloaded and installed"
-    test -e "/var/cache/snapd/icons-pool/${SNAP_LOCAL_ID}.icon"
-    test -e "/var/cache/snapd/icons-pool/${SNAP_FALLBACK_ID}.icon"
-    test -e "/var/cache/snapd/icons/${SNAP_LOCAL_ID}.icon"
-    test -e "/var/cache/snapd/icons/${SNAP_FALLBACK_ID}.icon"
-
-    echo "Check that icons download and install directories are only writable by root"
-    # root must have wrx, but otherwise only care that non-root doesn't have write
-    stat --format %A "/var/cache/snapd/icons-pool" | MATCH -e 'drwx.-..-.'
-    stat --format %A "/var/cache/snapd/icons" | MATCH -e 'drwx.-..-.'
-
-    echo "Check that icons downloaded and installed icons are only writable by root"
-    stat --format %A "/var/cache/snapd/icons-pool/${SNAP_LOCAL_ID}.icon" | MATCH -e '-rw-.--.--'
-    stat --format %A "/var/cache/snapd/icons-pool/${SNAP_FALLBACK_ID}.icon" | MATCH -e '-rw-.--.--'
-    stat --format %A "/var/cache/snapd/icons/${SNAP_LOCAL_ID}.icon" | MATCH -e '-rw-.--.--'
-    stat --format %A "/var/cache/snapd/icons/${SNAP_FALLBACK_ID}.icon" | MATCH -e '-rw-.--.--'
-
+    if [ "$SPREAD_VARIANT" = "with_icon_fallback" ]; then
+        echo "Check that installed snaps now have icons downloaded and installed"
+        test -e "/var/cache/snapd/icons-pool/${SNAP_LOCAL_ID}.icon"
+        test -e "/var/cache/snapd/icons-pool/${SNAP_FALLBACK_ID}.icon"
+        test -e "/var/cache/snapd/icons/${SNAP_LOCAL_ID}.icon"
+        test -e "/var/cache/snapd/icons/${SNAP_FALLBACK_ID}.icon"
+
+        echo "Check that icons downloaded and installed icons are only writable by root"
+        stat --format %A "/var/cache/snapd/icons-pool/${SNAP_LOCAL_ID}.icon" | MATCH -e '-rw-.--.--'
+        stat --format %A "/var/cache/snapd/icons-pool/${SNAP_FALLBACK_ID}.icon" | MATCH -e '-rw-.--.--'
+        stat --format %A "/var/cache/snapd/icons/${SNAP_LOCAL_ID}.icon" | MATCH -e '-rw-.--.--'
+        stat --format %A "/var/cache/snapd/icons/${SNAP_FALLBACK_ID}.icon" | MATCH -e '-rw-.--.--'
+
+        echo "Check that icons download and install directories are only writable by root"
+        # root must have wrx, but otherwise only care that non-root doesn't have write
+        stat --format %A "/var/cache/snapd/icons-pool" | MATCH -e 'drwx.-..-.'
+        stat --format %A "/var/cache/snapd/icons" | MATCH -e 'drwx.-..-.'
+      fi
```

I think it would be good to test that icons were *not* downloaded if a cloud is in use.

```suggestion
    else
        echo "Check that icons were not downloaded"
        test ! -e "/var/cache/snapd/icons-pool/${SNAP_LOCAL_ID}.icon"
        test ! -e "/var/cache/snapd/icons-pool/${SNAP_FALLBACK_ID}.icon"
        test ! -e "/var/cache/snapd/icons/${SNAP_LOCAL_ID}.icon"
        test ! -e "/var/cache/snapd/icons/${SNAP_FALLBACK_ID}.icon"
    fi
```